sdk-springboot/src/main/java/io/dapr/springboot/DaprTopicSubscription.java

```diff
@@ -20,64 +20,78 @@
  * Class to represent a subscription topic along with its metadata.
  */
 class DaprTopicSubscription {
+
   private final String pubsubName;
   private final String topic;
   private final String route;
+  private final String deadLetterTopic;
   private final DaprTopicRoutes routes;
   private final Map<String, String> metadata;
   private final DaprTopicBulkSubscribe bulkSubscribe;
 
   /**
    * Create a subscription topic.
-   * @param pubsubName The pubsub name to subscribe to.
-   * @param topic The topic to subscribe to.
-   * @param route Destination route for messages.
-   * @param metadata Metdata for extended subscription functionality.
+   *
+   * @param pubsubName      The pubsub name to subscribe to.
+   * @param topic           The topic to subscribe to.
+   * @param route           Destination route for messages.
+   * @param deadLetterTopic Name of topic to forward undeliverable messages.
+   * @param metadata        Metdata for extended subscription functionality.
    */
-  DaprTopicSubscription(String pubsubName, String topic, String route, Map<String, String> metadata) {
-    this(pubsubName, topic, route, metadata, null);
+  DaprTopicSubscription(String pubsubName, String topic, String route, String deadLetterTopic,
```

Comment:
Please, keep these small for simpler scenarios. For complete params, just add to the last one. Optionally, create new variations by adding deadletterTopic, but don't change existing ones.